src/contracts/facilitators/gsm/swapFreezer/OracleSwapFreezer.sol

```diff
@@ -86,7 +86,7 @@ contract OracleSwapFreezer is AutomationCompatibleInterface {
   }
 
   /// @inheritdoc AutomationCompatibleInterface
-  function checkUpkeep(bytes calldata) external view returns (bool, bytes memory) {
+  function checkUpkeep(bytes calldata) public view virtual returns (bool, bytes memory) {
```

Comment:
let's create a base (abstract) oracle swap freezer and inherit to create the CLL and Gelato ones
`ChainlinkOracleSwapFreezer` and `GelatoOracleSwapFreezer`